web/src/util.js

```diff
@@ -1,5 +1,99 @@
-const example = x => x
+const COLOR_CLS = { num: 'crimson', null: 'blue', bool: 'purple', str: 'maroon',  key: 'teal', default: 'black'}
 
+const typeToClass = {
+    'boolean': COLOR_CLS.bool,
+    'number': COLOR_CLS.num,
+    'string': COLOR_CLS.str,
+    'object': COLOR_CLS.default,
+}
+
+/**
+ * Returns the stringified JSON with Syntax Highlighting
+ * @param {object | Array} myJson The Object to stringify.
+ * @returns {string} The stringified Object.
+ */
+const getStringifiedJSON = (myJson, indentation) => {
+    if (myJson === undefined) {
+        return undefined
+    }
+    const stringifiedJSON = JSON.stringify(myJson, null, indentation)
+    let lines = stringifiedJSON.split('\n')
+    lines = lines.map(line => {
+        //These are the indexes where a Quotiation Escape happens
+        const escapeIndexes = [...line.trim().matchAll(/\\"/gm)].map(a => a.index)
```

Comment:
Das würde aber auch einen Escaped-Backslash am Ende matchen, der keine Quotiation-Escape ist: `"\\"` ...das kannst du bestimmt noch einbauen, dass auch das abgefangen wird.

Ich will dir aber trotzdem noch mal zeigen, wie ich das wohl mit Parsing ungefähr angegangen wäre (wenn du Zeit und Lust hast, kannst du das ja noch als 2. Variante probieren fertig zu schreiben und dann können wir über beide die Tests jagen. Musst du aber nicht machen, du hast bestimmt ab Morgen andere Sorgen. Guten Flug btw!). Ist am Ende natürlich etwas umfangreicher als deine Lösung, dafür lässt sich einfacher die Korrektheit sicherstellen.
```
const extractBool = input => {
    if (input.toLowerCase().startsWith('true'))
	return input.substring(0, 4)

    if (input.toLowerCase().startsWith('false'))
	return input.substring(0, 5)

    throw Error(`Expected bool token but got "${input}"`)    
}

// ...weitere Token-Extractor

const getToken = input => {
    const char = input[0].toLowerCase()
    if (char === '"') {	
	return {
           type: 'string',
	   raw: extractString(input)
        }
    } else if (['t', 'f'].includes(char)) {
    	return {
           type: 'bool',
	   raw: extractBool(input)
        }
    } else if (/[.0-9]/.test(char)) {
	return {
	   type: 'number',
	   raw: extractNumber(input)
        }
    } else if (['{', '}', '[', ']', ':'].includes(char)) {
	return {
	   type: 'symbol',
	   raw: input[0]
        } 
    } else if (char === 'n') {
 	return {
           type: 'null',
	   raw: extractNull(input)
	}
    } else if (['', "\n", "\t"].includes(char)) {
	return {
	   type: 'space',
	   raw: extractSpaces(input)
        }
    }
    throw Error(`Could not parse line: "${input}"`)
}

const parseNextToken = input => {
    if (!input) return null

    const token = getToken(input)
    return [ token, token ? input.substring(token.raw.length) : '']
}

const getStringifiedJSON = (input, indentation) => {

    const stringified = JSON.stringify(input, null, indentation)
    if (stringified === undefined) 
         return undefined

    const lines = stringified.split('\n')    
    for (const line of lines) {
       const lineTokens = []
       let remaining = input
       while (remaining) {
         let [ token, remaining ] = parseNextToken(input)
         if (token) lineTokens.push(token)
       }

       let newLine = ''
       for (const token of lineTokens) {
          const cls = type2class[token.type]
          newLine += cls ? `<span class="${cls}">${token.raw}</span>` : token.raw;
       }
       newLines.push(newLine)
    }
    return newLines.join('/n')
}
````

Reply:
Verdammt. Der Quotation Escape am ende machts halt echt kaputt. Auf Anhieb fällt mir jetzt auch keine schöne Lösung ein das abzufangen ohne den ganzen spaß jetzt unnötig zu verkomplizieren. 

Ich denke, ich schreibe deine Lösung mal im Flieger runter und schau mal wie dat klappt.

Man könnte natürlich auch eine Lösung vom Internet ziehen, die Absurd kurz ist und auch echt zu funktionieren scheint. Habe nicht extrem durchgetestet, aber ich check halt die regex nur so mäßig. Weiß nicht wie bulletproof das ist

https://codepen.io/absolutedevelopment/pen/EpwVzN

Reply:
Das ist in der Tat recht kompakt. Wenn wir die Tests haben, können wir da mal die Implementierungen dagegenschmeissen. Ich werde am Wochende auch mal ein paar zusätzliche Tests beisteuern.

Reply:
Das schöne ist, dass er da schon richtig robuste Regexps für die verschiedenen Tokentypen drinstehen hat - bei Number behandelt er auch Exponentialnotation, was bei uns fehlt. Zudem wandelt er auch `<, >, &` in Html-Entitäten um - das müssten wir wahrscheinlich auch machen